app/uk/gov/hmrc/gvmshaulierstatus/services/HaulierStatusService.scala

```diff
@@ -66,14 +66,14 @@ class HaulierStatusService @Inject() (
 
   def updateStatus()(implicit headerCarrier: HeaderCarrier): Future[Unit] =
     haulierStatusRepository.findAllOlderThan(appConfig.intervalSeconds, appConfig.limit).flatMap { documents =>
-      val receivedDocsPercentage = if (documents.nonEmpty) (documents.count(_.status == Received).toFloat / documents.size) * 100 else 0
+      val receivedDocsPercentage = if (documents.nonEmpty) (documents.count(_.status.value === Received.value).toFloat / documents.size) * 100 else 0
```

Comment:
This looks like a big step backwards - we don't want to replace a type safe match with a String match.

Reply:
updated.